app/services/link_agency_identities.rb

```diff
@@ -0,0 +1,55 @@
+class LinkAgencyIdentities
+  AGENCY_INFO = Struct.new(:issuer, :priority, :agency_id)
+
+  def link
+    sps = sps_to_link
+    sps.sort! { |spa, spb| spa.priority <=> spb.priority }
+    sps.each { |sp| link_service_provider(sp.agency_id, sp.issuer) }
+    self
+  end
+
+  def self.report
+    report_sql = <<~SQL
+      SELECT DISTINCT ag.name, id.uuid as old_uuid, ai.uuid as new_uuid
+      FROM agency_identities ai, service_providers sp, identities id, agencies ag
+      WHERE ag.id = ai.agency_id AND ag.id = sp.agency_id AND ai.user_id = id.user_id and
```

Comment:
is this `and` lowercase for a reason?

Reply:
No. I should probably make it consistent.  Generally I make SQL capital and everything else small.  Do you have a preferred convention?